internal/service/subscription.go

```diff
@@ -333,19 +334,22 @@ func (s *subscriptionService) CreateSubscription(ctx context.Context, req dto.Cr
 		}
 
 		// Create invoice for the subscription (in case it has advance charges)
-		invoiceService := NewInvoiceService(s.ServiceParams)
 		_, err = invoiceService.CreateSubscriptionInvoice(ctx, &dto.CreateSubscriptionInvoiceRequest{
 			SubscriptionID: sub.ID,
 			PeriodStart:    sub.CurrentPeriodStart,
 			PeriodEnd:      sub.CurrentPeriodEnd,
 			ReferencePoint: types.ReferencePointPeriodStart,
 		})
+
```

Comment:
need handling here to make sure that in case there is no invoice or the invoice is 0$ then in that case we activate the subscription here itself